dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/utils/McpClient.java

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.fat.utils;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import org.junit.rules.ExternalResource;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpRequest;
+
+/**
+ * Initializes an MCP session before each test.
+ * Sends an 'initialize' request and stores the returned session ID from the response headers.
+ * Also sends a 'notifications/initialized' notification to complete handshake.
+ * Make sure to add "String response = client.callMCP(request);"
+ */
+
+/**
+ * Example of use:
+ * Make sure you have @Rule annotaition and add initialize the McpClient
+ * public McpClient client = new McpClient(server, "/[whatever your root file is named]");
+ *
+ * 
+ */
+public class McpClient extends ExternalResource {
```

Comment:
```suggestion
/**
 * A client for testing the MCP server, which takes care of setting up a session, passing
 * through the correct headers on each request and deleting the session after the test.
 * <p>
 * This class must be initialized as a JUnit test rule in a field in the test class:
 * <pre>{@code
 * @Rule
 * public McpClient client = new McpClient(server, "/[whatever your root file is named]");
 * }</pre>
 * <p>
 * Then within your test method, you can use the client to make requests:
 * <pre>{@code
 * String jsonResponse = client.callMCP(jsonRequest);
 * }</pre>
 */
public class McpClient extends ExternalResource {
```

I think I have the Javadoc formatting right, but please check that Eclipse renders it correctly in the pop-up when you hover on the class name.

Reply:
You have the right Javadoc formatting, pop-up shows right format